entities/schema/collection.go

```diff
@@ -0,0 +1,367 @@
+// with default initialized structs in Property.
+func (p Property) ToModel() models.Property {
+	var m models.Property
+
+	m.DataType = p.DataType
+	m.Description = p.Description
+	indexFilterable := p.IndexFilterable
+	m.IndexFilterable = &indexFilterable
+	indexInverted := p.IndexInverted
+	m.IndexInverted = &indexInverted
+	indexSearchable := p.IndexSearchable
+	m.IndexSearchable = &indexSearchable
+	m.ModuleConfig = p.ModuleConfig
+	m.Name = p.Name
+	m.Tokenization = p.Tokenization
+	if len(p.NestedProperties) > 0 {
+		m.NestedProperties = make([]*models.NestedProperty, 0, len(p.NestedProperties))
+		for _, np := range p.NestedProperties {
+			npm := np.ToModel()
+			m.NestedProperties = append(m.NestedProperties, &npm)
+		}
+	}
+
+	return m
+}
+
+// ShardingConfigFromModel returns a ShardingConfig copied from m.
+// If m isn't a sharding.Config underneath the interface{}, a default initialized ShardingConfig will be returned.
+func ShardingConfigFromModel(m interface{}) ShardingConfig {
+	sc := ShardingConfig{}
+
+	v, ok := m.(sharding.Config)
+	if !ok {
+		return sc
+	}
+
+	sc.ActualCount = v.ActualCount
+	sc.ActualVirtualCount = v.ActualVirtualCount
+	sc.DesiredCount = v.DesiredCount
+	sc.DesiredVirtualCount = v.DesiredVirtualCount
+	sc.Function = v.Function
+	sc.Key = v.Key
+	sc.Strategy = v.Strategy
+	sc.VirtualPerPhysical = v.VirtualPerPhysical
+
+	return sc
+}
+
+// ToModel returns an interface{} containing a sharding.Config from s.
+func (s ShardingConfig) ToModel() interface{} {
+	var m sharding.Config
+
+	m.ActualCount = s.ActualCount
+	m.ActualVirtualCount = s.ActualVirtualCount
+	m.DesiredCount = s.DesiredCount
+	m.DesiredVirtualCount = s.DesiredVirtualCount
+	m.Function = s.Function
+	m.Key = s.Key
+	m.Strategy = s.Strategy
+	m.VirtualPerPhysical = s.VirtualPerPhysical
+
+	return m
+}
+
+// CollectionFromClass returns a Collection copied from m.
+func CollectionFromClass(m models.Class) Collection {
+	c := Collection{}
+
+	c.Name = m.Class
+	c.Description = m.Description
+	if m.InvertedIndexConfig != nil {
+		c.InvertedIndexConfig = InvertedIndexConfigFromModel(*m.InvertedIndexConfig)
+		if v, ok := m.ModuleConfig.(map[string]interface{}); ok {
+			c.ModuleConfig = v
+		}
+	}
+	if m.MultiTenancyConfig != nil {
+		c.MultiTenancyConfig.Enabled = m.MultiTenancyConfig.Enabled
+	}
+	c.Properties = make([]Property, len(m.Properties))
+	for i, mp := range m.Properties {
+		p := PropertyFromModel(*mp)
+
+		c.Properties[i] = p
+	}
+	if m.ReplicationConfig != nil {
+		c.ReplicationConfig.Factor = m.ReplicationConfig.Factor
+	}
+	if m.ShardingConfig != nil {
+		c.ShardingConfig = ShardingConfigFromModel(m.ShardingConfig)
+	}
+	c.VectorIndexType = stringToVectorIndexType[m.VectorIndexType]
+	switch c.VectorIndexType {
+	case VectorIndexTypeHNSW:
+		c.VectorIndexConfig = m.VectorIndexConfig.(hnsw.UserConfig)
+	case VectorIndexTypeOther:
```

Comment:
As of now, the only supported index is HNSW. I'm curious about whether 'other' is a valid index option

Reply:
I remove it. Indeed it doesn't serve a purpose.